lib/ht_members.rb

```diff
@@ -1,17 +1,66 @@
 # frozen_string_literal: true
 
-# WIP: cache of information about HathiTrust members
+require "mysql2"
+require "services"
+
+# Information about an individual HathiTrust institution
+class HTMember
+  attr_reader :inst_id, :country_code, :weight, :oclc_sym
+
+  def initialize(inst_id:, country_code: nil, weight: nil, oclc_sym: nil)
+    @inst_id = inst_id
+    @country_code = country_code
+    @weight = weight
+    @oclc_sym = oclc_sym
+  end
+end
+
+=begin
+
+Cache of information about HathiTrust members.
+
+Usage:
+
+  htm = HTMembers.new()
+  cc = htm["yale"].country_code
+  wt = htm["harvard"].weight
+
+This returns a hash keyed by member id that contains the country code, weight,
+and OCLC symbol.
+
+You can also pass in mock data for development/testing purposes:
+
+  htm = HTMembers.new({
+    "haverford" => HTMember.new(inst_id: "haverford", country_code: "us", weight: 0.67)
+  })
+  htm["haverford"].country_code
+  htm["haverford"].weight
+
+=end
+
 class HTMembers
 
-  def initialize(*members)
-    @members = members.empty? ? load_from_db : members
+  attr_reader :members
+
+  def initialize(members = load_from_db)
+    @members = members
   end
 
   def load_from_db
-    # do that
+    Services.holdings_db[:ht_institutions]
+      .select(:inst_id,:country_code,:weight,:oclc_sym)
+      .as_hash(:inst_id)
+      .transform_values { |h| HTMember.new(h) }
   end
 
-  def get_list
-    @members
+  # Given a inst_id, returns a hash of data for that member.
+  def [](inst_id)
+    if @members.key?(inst_id) then
+      @members[inst_id]
+    else
+      raise "No member_info data for inst_id:#{inst_id}"
```

Comment:
Might want to change this back to a warning

Reply:
Could just remove the warning altogether?

Reply:
Could also add the new member to the cached member data when doing an estimate

Reply:
I say leave it as an error. If we want to pretend some institution is a partner, we should change it at the source layer. That means HTMember needs `#temp_add` or something. I'd resist just calling it `add` or `<<`  -- make it clear it's not changing the underlying datastore.

Reply:
Sounds good. Pushed a commit with this.